### PR TITLE
Radtts ragged

### PR DESCRIPTION
**Pull request by borisfom**
# What does this PR do ?

Adding ragged batch input option for RadTTS - with some refactoring
Fixing 2 bugs from previous PR: 
   1. clamp(1,..)
   2. argsort() for unsort_tensor() was in wrong direction  

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: tests/collections/nlp/test_nmt_model.py [hunk 1]
```diff
@@ -110,7 +110,7 @@ def test_train_eval_loss(self):
         eval_loss = model.eval_loss_fn(log_probs=log_probs, labels=tgt_ids)
         assert torch.allclose(train_loss, eval_loss)
 
-    @pytest.mark.skipif(not os.path.exists('/home/TestData/nlp'), reason='Not a Jenkins machine')
+    # @pytest.mark.skipif(not os.path.exists('/home/TestData/nlp'), reason='Not a Jenkins machine')
```
Comment (blisc) [inline tests/collections/nlp/test_nmt_model.py:113]: What was this from?

File: nemo/collections/tts/helpers/helpers.py [hunk 2]
```diff
@@ -706,3 +706,87 @@ def mask_sequence_tensor(tensor: torch.Tensor, lengths: torch.Tensor):
...
+
+    index = 1
+    num_batches = batch_lengths.shape[0] - 1
+    texts = torch.zeros(num_batches, max_len, dtype=torch.int64, device=text.device) + padding_idx
+    pitches = torch.ones(num_batches, max_len, dtype=torch.float32, device=text.device)
+    paces = torch.zeros(num_batches, max_len, dtype=torch.float32, device=text.device) + 1.0
+    volumes = torch.zeros(num_batches, max_len, dtype=torch.float32, device=text.device) + 1.0
+    lens = torch.zeros(num_batches, dtype=torch.int64, device=text.device)
+    speakers = torch.zeros(num_batches, dtype=torch.int64, device=text.device)
+    last_index = index - 1
```
Comment (blisc) [inline nemo/collections/tts/helpers/helpers.py:732]: Why do we need `last_index`?

File: nemo/collections/tts/helpers/helpers.py [hunk 3]
```diff
@@ -706,3 +706,87 @@
...
+    max_len = torch.max(batch_lengths[1:] - batch_lengths[:-1])
+
+    index = 1
+    num_batches = batch_lengths.shape[0] - 1
+    texts = torch.zeros(num_batches, max_len, dtype=torch.int64, device=text.device) + padding_idx
+    pitches = torch.ones(num_batches, max_len, dtype=torch.float32, device=text.device)
+    paces = torch.zeros(num_batches, max_len, dtype=torch.float32, device=text.device) + 1.0
+    volumes = torch.zeros(num_batches, max_len, dtype=torch.float32, device=text.device) + 1.0
+    lens = torch.zeros(num_batches, dtype=torch.int64, device=text.device)
+    speakers = torch.zeros(num_batches, dtype=torch.int64, device=text.device)
```
[the remark this answers is not in the document]
Comment (blisc) [reply nemo/collections/tts/helpers/helpers.py:731]: Let's remove speaker from this function

File: nemo/collections/tts/helpers/helpers.py (hunk 2 above)
Comment (borisfom) [reply nemo/collections/tts/helpers/helpers.py:732]: To avoid repeating index-1.

**Comment by borisfom** [comment]
Superseded by #6020

**Pull request closed without merging**